src/bsoncxx/array/view.cpp

```diff
@@ -47,19 +47,15 @@ view::iterator& view::iterator::operator++() {
     }
 
     bson_iter_t i;
-    i.raw = _element.raw;
-    i.len = _element.length;
-    i.next_off = _element.offset;
+    i.raw = _element.raw();
```

Comment:
We have this code four times. Is there a way we could just have a private to_bson_iter_t method?